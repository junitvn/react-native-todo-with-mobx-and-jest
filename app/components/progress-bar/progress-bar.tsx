import * as React from "react"
import { StyleProp, View, ViewStyle } from "react-native"
import { observer } from "mobx-react-lite"

const CONTAINER: ViewStyle = {
  justifyContent: "flex-start",
  height: 3,
  backgroundColor: "lightgray",
  borderRadius: 3,
}

const PROGRESS: ViewStyle = {
  height: 3,
  borderRadius: 3,
}

export interface ProgressBarProps {
  current: number
  total: number
  color: string
  style?: StyleProp<ViewStyle>
}

export const ProgressBar = observer(function ProgressBar(props: ProgressBarProps) {
  const { style, current, total, color } = props
  const styles = Object.assign({}, CONTAINER, style)
  const progress = (current / total) * 100

  return (
    <View style={styles}>
      <View style={[PROGRESS, { width: `${progress}%`, backgroundColor: color }]} />
    </View>
  )
})
