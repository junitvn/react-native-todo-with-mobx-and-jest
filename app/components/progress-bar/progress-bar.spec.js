import React from "react"
import renderer from "react-test-renderer"
import { ProgressBar } from "./progress-bar"

test("renders correctly", () => {
  const tree = renderer.create(<ProgressBar />).toJSON()
  expect(tree).toMatchSnapshot()
})
