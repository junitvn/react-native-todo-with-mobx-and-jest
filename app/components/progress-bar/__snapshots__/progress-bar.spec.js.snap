// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "lightgray",
      "borderRadius": 3,
      "height": 3,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "height": 3,
        },
        Object {
          "backgroundColor": undefined,
          "width": "NaN%",
        },
      ]
    }
  />
</View>
`;
